Cyberpunk2k20_v2/Cyberpunk2k20.html

```diff
@@ -0,0 +1,3893 @@
+            <button class="skill-button" type="roll" value="&amp;{template:skill}{{name=@{character_name}}}{{skillname=Expert: High Tech}} {{result=[[1d10!cf<1 + @{experthightech} + @{experthightech_mod}+ @{int} + @{int_mod} - @{int_penality} + ?{mod.|0}]] }}{{mod=?{mod.|0}}}{{fumble=[[{1d10}<@{experthightech_max}]]}}" name="roll_skillRoll">
+                <b>E. High Tech</b>
+                <span name="attr_experthightech_max"></span>
+            </button>
+            <b><i>Expert: High Tech</i><input type="number" name="attr_experthightech" value="0" maxlength="2">
+                <input type="number" name="attr_experthightech_mod" value="0" maxlength="2">
+                <input type="hidden" name="attr_experthightech_max" value="0">
+            </b>
+            <input type="checkbox" class="skill-select" name="attr_expertintelligenceanalysis_select">
+            <button class="skill-button" type="roll" value="&amp;{template:skill}{{name=@{character_name}}}{{skillname=Expert: Intelligence Analysis}} {{result=[[1d10!cf<1 + @{expertintelligenceanalysis} + @{expertintelligenceanalysis_mod}+ @{int} + @{int_mod} - @{int_penality} + ?{mod.|0}]] }}{{mod=?{mod.|0}}}{{fumble=[[{1d10}<@{expertintelligenceanalysis_max}]]}}" name="roll_skillRoll">
+                <b>E. Intelligence Analysis</b>
+                <span name="attr_expertintelligenceanalysis_max"></span>
+            </button>
+            <b><i>Expert: Intelligence Analysis</i><input type="number" name="attr_expertintelligenceanalysis" value="0" maxlength="2">
+                <input type="number" name="attr_expertintelligenceanalysis_mod" value="0" maxlength="2">
+                <input type="hidden" name="attr_expertintelligenceanalysis_max" value="0">
+            </b>
+            <input type="checkbox" class="skill-select" name="attr_expertinternationalbusiness_select">
+            <button class="skill-button" type="roll" value="&amp;{template:skill}{{name=@{character_name}}}{{skillname=Expert: International Business}} {{result=[[1d10!cf<1 + @{expertinternationalbusiness} + @{expertinternationalbusiness_mod}+ @{int} + @{int_mod} - @{int_penality} + ?{mod.|0}]] }}{{mod=?{mod.|0}}}{{fumble=[[{1d10}<@{expertinternationalbusiness_max}]]}}" name="roll_skillRoll">
+                <b>E. International Business</b>
+                <span name="attr_expertinternationalbusiness_max"></span>
+            </button>
+            <b><i>Expert: International Business</i><input type="number" name="attr_expertinternationalbusiness" value="0" maxlength="2">
+                <input type="number" name="attr_expertinternationalbusiness_mod" value="0" maxlength="2">
+                <input type="hidden" name="attr_expertinternationalbusiness_max" value="0">
+            </b>
+            <input type="checkbox" class="skill-select" name="attr_expertkamasutra_select">
+            <button class="skill-button" type="roll" value="&amp;{template
```

Comment:
This Javascript could be cleaned up with [forEach loops](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/forEach). Something like this:

```
["body","body_mod","endurance","endurance_mod"].forEach(attr => {
   	on(`change:${attr}`, () => {
   		const ability = (attr.includes("mod")) ? attr.split("_mod")[0] : attr;
   		getAttrs([`${ability}`,`${ability}_mod`, `${ability}_penalty`], (v) => {
			setAttrs({
			  [`${ability}_max`]: parseInt(v[`${ability}`])+parseInt(v[`${ability}_mod`])-parseInt(v[`${ability}_penalty`])
			});
		});
	});
});
```

Reply:
Sorry I don t understand what you mean about that. If I use Js to build my sheet it's become an API feature ?

Reply:
Aaah !!!! I see what you mean !!! I did know that I can used forEach loop in the worker. I know it's like a html website but I thought that roll20 need separate script. Ok ... I will work to simplify my worker script. But I could take some time because I m returning to my professionnal activity.